dsl/src/commonTest/kotlin/it/unibo/collektive/aggregate/SharingTest.kt

```diff
@@ -83,19 +83,20 @@ class SharingTest : StringSpec({
         // Device 1
         val testNetwork1 = NetworkImplTest(nm, id1)
         aggregate(id1, condition, testNetwork1) {
-            val res1 = share(initV1) { f ->
-                f.toMap().maxBy { it.value }.value
+            val res1 = share(initV1) {
+                it.maxBy { v -> v.value }.value
             }
             res1 shouldBe initV1
 
-            val res2 = share(initV1) { f ->
-                f.toMap().maxBy { it.value }.value butReturn "A string"
+            val res2 = share(initV1) {
+                val min = it.maxBy { v -> v.value }.value
+                min yielding "A string"
             }
             res2 shouldBe "A string"
 
-            val res3 = share(initV1) { f ->
-                val min = f.toMap().minBy { it.value }.value
-                min butReturn if (min > 1) "Hello" else null
+            val res3: String? = share(initV1) {
```

Comment:
Here works also without the type annotation `String?`?

Reply:
Nope, still got problems with overload